ui-react/src/components/exporters/EditGoogleSheetsExporter.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Alert, Button, Card, Form
+} from 'react-bootstrap';
+import { Prev } from 'react-bootstrap/esm/PageItem';
+import {
+  googleLogin,
+  validateToken,
+  createSpreadsheet,
+} from '../../eventsBridge';
+import {
+  Credentials,
+  Exporter,
+  GoogleSheetsConfig,
+  OutputVendorName,
+} from '../../types';
+import SheetsCombobox from './SheetsCombobox';
+import styles from './EditGoogleSheetsExporter.module.css';
+
+enum Status {
+  LOADING,
+  LOGGED_IN,
+  LOGIN,
+  ERROR,
+}
+
+type EditGoogleSheetsExporterProps = {
+  handleSave: (exporter: Exporter) => Promise<void>;
+  exporter: Exporter;
+};
+
+function EditGoogleSheetsExporter({
+  exporter,
+  handleSave,
+}: EditGoogleSheetsExporterProps) {
+  const [status, setStatus] = useState<Status>(Status.LOADING);
+  const [errorMessage, SetErrorMessage] = useState();
+  const [readyToSave, setReadyToSave] = useState(false);
+
+  // const dataToReturn = setupExporterConfigForm(OutputVendorName.GOOGLE_SHEETS);
+  const [exporterConfig, setExporterConfig] = useState<GoogleSheetsConfig>(
+    exporter as GoogleSheetsConfig
+  );
+
+  useEffect(() => {
+    const validate = async () => {
+      // const valid = await validateToken(exporterConfig?.options?.credentials);
+      // setStatus(valid ? Status.LOGGED_IN : Status.LOGIN);
+    };
+
+    try {
+      validate();
+    } catch (ex) {
+      setStatus(Status.ERROR);
+      SetErrorMessage(ex.message);
+    }
+  }, [exporterConfig.options.credentials]);
+
+  const login = async () => {
+    try {
+      setStatus(Status.LOADING);
+
+      const credentials = await googleLogin();
+      setExporterConfig((prevExport) => ({
+        ...prevExport,
+        options: { ...prevExport.options, credentials },
+      }));
+
+      setStatus(Status.LOGGED_IN);
+    } catch (ex) {
+      setStatus(Status.ERROR);
+      SetErrorMessage(ex.message ?? ex);
+    }
+  };
+
+  const save = async () => {
+    const isNewSheet = exporterConfig.options.spreadsheetId.length < 30;
```

Comment:
How does this mean that it is a new sheet?
I suggest adding a comment why length < 30 means new sheet